HW0/P4/P4.py

```diff
@@ -30,47 +34,86 @@
     # Load true trajectory and plot it
     # Normally, this data wouldn't be available in the real world
     #####################
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '--b', label='True trajectory')
+    s_true = np.loadtxt('P4_trajectory.txt',delimiter=',')
+    x_true = s_true[:,0]
+    y_true = s_true[:,1]
+    z_true = s_true[:,2]
+    
+    ax.plot(x_true, y_true, z_true,
+            '--b', label='True trajectory')
 
     #####################
     # Part 2:
     #
     # Read the observation array and plot it (Part 2)
     #####################
+    s_obs = np.loadtxt('P4_measurements.txt',delimiter=',')
+    s_obs_pos = s_obs[:,0:3]
+    s_obs_pos_adjust = np.transpose(np.dot(np.linalg.inv(C),np.transpose(s_obs_pos)))
+    x_obs_pos_adjust = s_obs_pos_adjust[:,0]
+    y_obs_pos_adjust = s_obs_pos_adjust[:,1]
+    z_obs_pos_adjust = s_obs_pos_adjust[:,2]
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    ax.plot(x_obs_pos_adjust, y_obs_pos_adjust, z_obs_pos_adjust,
+            '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
+    A = np.zeros((6,6))
+    A[0,np.array([0,3])] = np.array([1,dt])
+    A[1,np.array([1,4])] = np.array([1,dt])
+    A[2,np.array([2,5])] = np.array([1,dt])
+    A[3,3] = 1 - c*dt
+    A[4,4] = 1 - c*dt
+    A[5,5] = 1 - c*dt  
+    a = np.zeros(6)
+    a[5] = g*dt
+    s = np.zeros((6,K))
 
     # Initial conditions for s0
+    s[np.array([2,3,4,5]),0] = np.array([2,15,3.5,4])
     # Compute the rest of sk using Eq (1)
+    for ii in range(1,K):
+    	s[:,ii] = np.dot(A,s[:,ii-1]) + a
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '-k', label='Blind trajectory')
+    ax.plot(s[0,:], s[1,:], s[2,:],
+            '-k', label='Blind trajectory')
 
     #####################
     # Part 4:
     # Use the Kalman filter for prediction
     #####################
 
-    # B = ?
-    # C = ?
+    B = np.zeros((6,6))
+    B[0,0] = bx
```

Comment:
Not that it really matters, but to keep code cleaner and to take advantage of numpy, you should try and consolidate things like this to take up less space and to be more concise. For example